backend/index.js

```diff
@@ -1,138 +1,201 @@
+        console.error('You need to define the new provider');
+        return callback();
+      }
+
+      pizzaParserClass = pizzaProvider.get(args.provider);
+      if (!pizzaParserClass) {
+        console.error('Provider is not known');
+        return callback();
+      }
+      pizzaParser = new pizzaParserClass();
+      appCtrl = new AppController(pizzaParser);
+
+      // We reload the app
+      loadServer();
+      return callback();
+    });
+  vorpal
+    .command('providers', 'List available providers')
+    .action((args, callback) => {
+      console.log('Here are the handsome providers available (active on is bracketed)');
+      const currentProvider = pizzaProvider.getNameFromClass(pizzaParserClass);
+      pizzaProvider.getAvailableProviders().map( provider => {
+        message = provider;
+        if (provider == currentProvider) {
+          message = '{' + message + '}';
         }
-      })
-
-      socket.on('disconnect', () => {
-        if (currentUser === null) {
-          return
-        }
-        appCtrl.getUsersModel().setUserOffline(currentUser)
-        if (appCtrl.getUsersModel().getNbConnectionsUser(currentUser) === 0) {
-          io.sockets.emit('DISCONNECT_USER_SUCCESS', currentUser.id)
-        }
-      })
-    })
-
-    appCtrl.getOrdersModel().setHourEnd(currentDate.getHours() + 1)
-    appCtrl.getOrdersModel().setMinuteEnd(currentDate.getMinutes())
-
-    if (!isServerStarted) {
-      isServerStarted = true
-      server.listen(3000)
-    }
-
-    vorpal
-      .command('countdown', 'Set or change when to stop the orders')
-      .option('--hour <hour>', 'Set or change the hour when to block the orders')
-      .option('--minute <minute>', 'Set or change the minute when to block the orders')
-      .action((args, callback) => {
-        if (typeof args.options.hour === 'undefined' || typeof args.options.minute === 'undefined') {
-          console.error('You need to define --hour and --minute')
-          return callback()
-        }
-
-        if (typeof args.options.hour !== 'number' || typeof args.options.minute !== 'number') {
-          console.error('"hour" and "minute" should be integers')
-          return callback()
-        }
-
-        const hourEnd = parseInt(args.options.hour)
-        const minuteEnd = parseInt(args.options.minute)
-
-        if (hourEnd < 0 || hourEnd > 23) {
-          console.error('"hour" must be between 0 and 23')
-          return callback()
-        }
-
-        if (minuteEnd < 0 || minuteEnd > 59) {
-          console.error('"minute" must be between 0 and 59')
-          return callback()
-        }
-
-        appCtrl.getOrdersModel().setHourEnd(hourEnd)
-        appCtrl.getOrdersModel().setMinuteEnd(minuteEnd)
-
-        io.sockets.emit('SET_COUNTDOWN', {
-          hour: appCtrl.getOrdersModel().getHourEnd(),
-          minute: appCtrl.getOrdersModel().getMinuteEnd()
-        })
+        console.log(message);
+      });
```

Comment:
```javascript
pizzaProvider.getAvailableProviders().map(providerName => {
  const message = providerName;

  if (providerName === currentProvider) {
    message = `{${message}}`;
  }

  console.log(`- ${message}`);
});
```

Reply:
Yeah, it is more clear like that. I've just declared "message" with "let" instead of "const" as far as we are updating the value ;)

Reply:
Yes my bad!